packages/arb-avm-cpp/avm/src/machinestate/machineoperation.cpp

```diff
@@ -1091,6 +1091,29 @@ void getbuffer256(MachineState& m) {
     ++m.pc;
 }
 
+uint64_t setbuffer_variable_gas_cost(MachineState const& m, uint64_t inc) {
+    if (m.stack.stacksize() < 3) {
+        return 0;
+    }
+    auto offset = std::get_if<uint256_t>(&m.stack[0]);
+    auto buf = std::get_if<Buffer>(&m.stack[2]);
+    if (!buf || !offset) {
+        return 0;
+    }
+    if (*offset + inc > std::numeric_limits<uint64_t>::max()) {
+        return 0;
```

Comment:
The instruction should have a defined non-zero ArbGas cost in error cases like this.

Reply:
In those cases the cost will be the fixed cost from `opcodes.hpp` just like for pairing...